finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/param.scala

```diff
@@ -1,11 +1,50 @@
 package com.twitter.finagle.buoyant.h2
 
 import com.twitter.finagle.Stack
+import com.twitter.util.StorageUnit
 
 package object param {
 
   case class ClientPriorKnowledge(assumed: Boolean)
   implicit object ClientPriorKnowledge extends Stack.Param[ClientPriorKnowledge] {
     val default = ClientPriorKnowledge(true)
   }
+
+  case class AutoRefillConnectionWindow(enabled: Boolean)
+  implicit object AutoRefillConnectionWindow extends Stack.Param[AutoRefillConnectionWindow] {
+    val default = AutoRefillConnectionWindow(false)
+  }
+
+  case class WindowUpdateRatio(ratio: Float)
+  implicit object WinodwUpdateRatio extends Stack.Param[WindowUpdateRatio] {
+    val default = WindowUpdateRatio(0.95f)
+  }
+
+  object Settings {
+
+    case class HeaderTableSize(size: Option[StorageUnit])
+    implicit object HeaderTableSize extends Stack.Param[HeaderTableSize] {
+      val default = HeaderTableSize(None)
+    }
+
+    case class InitialWindowSize(size: Option[StorageUnit])
+    implicit object InitialWindowSize extends Stack.Param[InitialWindowSize] {
+      val default = InitialWindowSize(None)
+    }
+
+    case class MaxConcurrentStreams(streams: Option[Long])
+    implicit object MaxConcurrentStreams extends Stack.Param[MaxConcurrentStreams] {
+      val default = MaxConcurrentStreams(None)
```

Comment:
```
 SETTINGS_MAX_CONCURRENT_STREAMS (0x3):  Indicates the maximum number
      of concurrent streams that the sender will allow.  This limit is
      directional: it applies to the number of streams that the sender
      permits the receiver to create.  Initially, there is no limit to
      this value.  It is recommended that this value be no smaller than
      100, so as to not unnecessarily limit parallelism.

      A value of 0 for SETTINGS_MAX_CONCURRENT_STREAMS SHOULD NOT be
      treated as special by endpoints.  A zero value does prevent the
      creation of new streams; however, this can also happen for any
      limit that is exhausted with active streams.  Servers SHOULD only
      set a zero value for short durations; if a server does not wish to
      accept requests, closing the connection is more appropriate.
```